src/plots/cartesian/axis_autotype.js

```diff
@@ -15,22 +15,26 @@ var Lib = require('../../lib');
 var BADNUM = require('../../constants/numerical').BADNUM;
 
 module.exports = function autoType(array, calendar, opts) {
-    opts = opts || {};
+    var convertNumeric = opts.convertNumeric;
 
     if(!opts.noMultiCategory && multiCategory(array)) return 'multicategory';
-    if(moreDates(array, calendar)) return 'date';
+    if(convertNumeric && moreDates(array, calendar)) return 'date';
     if(category(array)) return 'category';
-    if(linearOK(array)) return 'linear';
-    else return '-';
+    if(linearOK(array, convertNumeric)) return 'linear';
+    else return convertNumeric ? '-' : 'category';
```

Comment:
I think it would be better to leave `else return '-'`, for empty arrays and arrays with only `null` entries, and instead include `convertNumeric` in the `category(array)` call, ie replace:
```js
else if(Lib.cleanNumber(ai) !== BADNUM) curvenums++;
```
with something like:
```js
else if(convertNumeric ? Lib.cleanNumber(ai) !== BADNUM : typeof ai === 'number') curvenums++;
```

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/5240/commits/b73d3953825e977fa53c87d73faef1d07527671c.